wire-library/wire-tests/src/commonTest/proto/java/custom_options.proto

```diff
@@ -49,10 +49,22 @@ message FooBar {
 
   message More {
     repeated int32 serial = 1;
+
+    extend FooBar {
+      optional string more_string = 150;
+    }
+
+    option (my_message_option_six).(More.more_string) = "foobar";
```

Comment:
Can options actually be written as so? I thought the parens were used for custom options but what does it mean in this case? An option nested within another?

Reply:
Does protoc like this?

Reply:
Yes, this is valid. The path is a sequence of dot-separated names, and each name component can either be:
1. simple `field_name`
2. parenthesized `(optionally.qualified.extension.name)`

Examples:
* `option deprecated = true;`
   The field name is not parenthesized, so it's a normal field on the relevant options message.
* `option (my.message.option) = {};`
   The field name is parenthesized, so the whole thing (including all dot-separated components) is a reference to an extension (which might be qualified, hence the dots).
* `option (my.message.option).foo.bar.(Bar.baz) = 123;`
   In this case, `my.message.option` is an extension on an options message (a custom option). That extension's type is some message (let's call it `Foo`). The `foo` component is a normal field on `Foo` (let's say its type is a message named `Bar`). The `bar` component is a normal field on `Bar` (let's say its type is a message named `Baz`). Finally, the `Bar.baz` component refers to an _extension_ for `Baz`.

You can feel free to compile this file using `protoc custom_options.proto -o /dev/null` to see it is valid. If you want to see more details in the resulting descriptor, try the following:
```shell
protoc custom_options.proto -o /dev/stdout \
    | protoc --decode google.protobuf.FileDescriptorSet \
        google/protobuf/descriptor.proto custom_options.proto
```

Also see the docs for `google.protobuf.UninterpretedOption`: https://github.com/protocolbuffers/protobuf/blob/v21.5/src/google/protobuf/descriptor.proto#L740-L749

Reply:
Super cool, thank you